Core/include/Acts/Surfaces/SurfaceConcept.hpp

```diff
@@ -51,10 +51,10 @@ concept SurfaceConcept = requires(S s, const S cs, S s2, const S cs2,
     } -> std::same_as<void>;
   {
     cs.isOnSurface(gctx, Vector3{}, Vector3{},
-                   std::declval<const BoundaryCheck&>())
+                   std::declval<const BoundaryTolerance&>())
```

Comment:
Consider using the `requires` clause in L28/29 to create a value of type `const BoundaryTolerance&` instead of using declval.